### ISSUE TITLE
network detect output not the exactly same

### ISSUE THREAD

**Issue by milliyang** [ysh329/darknet2caffe#4]
I use the same  param (yolov2.cfg yolov2.weight   -> yolov2.prototxt, yolov2.caffemode) , to run the in two different network  (Darknet, and   https://github.com/eric612/Caffe-YOLOv2-Windows)

The final result are close, but are different !!! the box position and size are a little bit different too.

I have dump the network output of darknet, and hack it to Caffe-YOLOv2-Windows  right after forward() finish, to make sure the parsing of the final output format ( x,y,w,h,confidence,class) are correct. and they did works OK.

Finally, I dump all the layer output of darknet and Caffe-YOLOv2-Windows,  and I find that  even after the first convolution layer, the memory are different !!
(note: darknet convolution layer contains  convolution, batch normal, scale, and leak Rectify, but in caffe we use 4 layers)

Have you encounter this kinds of issue ?
how to solve it ? 
dig into the implement in convolution layer and do some modification?
do i have to train it in pure caffe project, not by convert the weight from darknet project?
the convertion from weight to caffemodel are not correct ?

any clue or suggestion is appreciate !

Thanks in advance!

**Comment by milliyang** [comment]
the left is caffe-yolov2-windows output and the right is darknet output:

<img width="1037" alt="caffe-yolov2-window-predict" src="https://user-images.githubusercontent.com/2309995/38170286-0e25b4ee-35b4-11e8-827b-c74288a26c88.png">

**Comment by ysh329** [comment]
You'd better check the input, first conv (feature map and filter params), first pool between darknet and caffe. 

[Here](https://github.com/ysh329/darknet)'s my version of darknet, only added some `printf`, easy to print out the conv filter param and feature map(s).

You can choose print-layer index using `print_idx` in file `src/network.c` as below or [here](https://github.com/ysh329/darknet/blob/master/src/network.c#L219):

```c
        int print_idx = -1;
        printf("== print_idx: %d ==\n", print_idx);
        int print_num = 10;
        if (net.index == print_idx)
            print_num = l.outputs;
        // check output
        for (int ii = 0; ii < print_num; ii++)
        {
            if (net.index == print_idx)
                printf("%d\t%f\n", ii, l.output[ii]);
            else
                fprintf(stderr, "%d\t%f\n", ii, l.output[ii]);
        }
```

Besides, [here](https://github.com/ysh329/common-scripts/blob/master/file/diff2.py)'s a script used to compare two equal-line-number files, such as feature map or conv params.

**Comment by ysh329** [comment]
If you want to print out conv filter params or feature map of Caffe, it's convenient to use pycaffe. I checked the correctness of conv params between darknet and converted models only for YOLOv2 and tiny-yolov, but I'm not sure how about yours. it's better checked again. Besides, some parameters such as in batchnorm or scale layers, affect the precision of final feature map.

**Comment by milliyang** [comment]
Thx, will try it.
